Issue template and contributing info (#576)
RI-SE/ATOS: docs/Contributing/Pull Requests/pull_requests.md

<a name="pull-requests"></a>
## Pull requests

**Please ask in the [discussions](https://github.com/RI-SE/ATOS/discussions) tab** before embarking on any significant pull request (e.g. implementing new features, refactoring code, porting to a different language),
otherwise you risk spending a lot of time working on something that the
project's developers might not want to merge into the project. 

> Please adhere to the coding conventions used throughout the project (indentation,

Author: ```suggestion
Please adhere to the [coding conventions](https://github.com/RI-SE/ATOS/wiki/Coding-Conventions) used throughout the project (indentation,
```

Author: Added a link to the coding conventions on the wiki

accurate comments, etc.) and any other requirements (such as test coverage).

Follow this process if you'd like your work considered for inclusion in the
project.

**IMPORTANT**: By submitting a patch, you agree to allow the project owner to
license your work under the same license as that used by the project.

#### Required information for a pull request

> * **What kind of change does this PR introduce?** (Bug fix, feature, docs update, etc.)
> 
> 
> * **What is the current behavior?** (You can also link to an open issue here)
> 
> 
> * **What is the new behavior (if this is a feature change)?**
> 
> 
> * **Does the PR change the way the software in this project interfaces with internal/external components such as files, other software, etc?**

Reviewer: ```suggestion
* What kind of change does this PR introduce? (Bug fix, feature, docs update, etc.)

* What is the current behavior? (You can also link to an open issue here)

* What is the new behavior (if this is a feature change)?

* Does the PR change the way the software in this project interfaces with internal/external components such as files, other software, etc?
```

#### Additional information

> * In the case that you feel there is additional information, files images or something else that will add valuable context, feel free to add anything extra that will help the developers understand the aim and purpouse of your pull request.**:

Reviewer: ```suggestion
* In the case that you feel there is additional information, files images, or something else that will add valuable context, feel free to add anything extra that will help the developers understand the aim and purpose of your pull request.:
```

> * **In the case that you feel there is additional information, files images or something else that will add valuable context, feel free to add anything extra that will help the developers understand the aim and purpouse of your pull request.**:

Reviewer: I realized when checking out the branch and reading the rendered text: If everything is bold-marked, then its difficult for the reader to understand what the important parts are.

> * **In the case that you feel there is additional information, files images or something else that will add valuable context, feel free to add anything extra that will help the developers understand the aim and purpouse of your pull request.**:

Author: Which parts do you think should not be in bold, when I look at the file only the headings and the required points are in bold. Previously the additional info was also in bold but I have changed that.

> * **In the case that you feel there is additional information, files images or something else that will add valuable context, feel free to add anything extra that will help the developers understand the aim and purpouse of your pull request.**:

Reviewer: ```suggestion
* In the case that you feel there is additional information, files images or something else that will add valuable context, feel free to add anything extra that will help the developers understand the aim and purpouse of your pull request.:
```

<a name="pull-requests"></a>
## Pull requests

> Good pull requests - patches, improvements, new features - are great.

Reviewer: Good pull requests are great? This sentence does not add a lot of content.

> Good pull requests - patches, improvements, new features - are great.

Author: ```suggestion
```
Agreed, might be a bit useless.

> Good pull requests - patches, improvements, new features - are great.

Author: ```suggestion
```

**Please ask first** before embarking on any significant pull request (e.g.
implementing new features, refactoring code, porting to a different language),
otherwise you risk spending a lot of time working on something that the
project's developers might not want to merge into the project. 

> [The discussions tab](https://github.com/RI-SE/ATOS/discussions) on the repo can be for this if you want to make sure that your endevour is something the maintainers consider relevant for the project.

Reviewer: ```suggestion
**Please ask in the [discussions](https://github.com/RI-SE/ATOS/discussions) tab** before embarking on any significant pull request (e.g. implementing new features, refactoring code, porting to a different language),
otherwise you risk spending a lot of time working on something that the
project's developers might not want to merge into the project. 
```

Please adhere to the coding conventions used throughout the project (indentation,

> accurate comments, etc.) and any other requirements (such as test coverage).

Reviewer: Where can these conventions be found?

Reviewer: Our conventions for branch-naming is feature_xxxx and fix_xxxx afaik, I don't know if this is stated somewhere on the repo?

Author: Yes this was one of the points I wanted to discuss with the entire team. We do have some conventions that have been established beforehand, but I dont think we for example have a document at the moment with everything mentioned above. I was thinking of writing one but we should reach a consensus on what we want to use.

Reviewer: I suggest writing one first and then we can reach a consensus / improve upon it through the GitHub review function! 
I know that we have coding conventions on the wiki page, so you can link to that one for now.

Author: Yeah, I was thinking the same, using that as a base. I will add that.

Follow this process if you'd like your work considered for inclusion in the
project.

**IMPORTANT**: By submitting a patch, you agree to allow the project owner to
license your work under the same license as that used by the project.

#### Required information for a pull request

> * **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

Reviewer: ```suggestion
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, etc.)
```

> * **What is the current behavior?** (You can also link to an open issue here)

Reviewer: Why do they need to describe the current behavior?

Author: This was a recommended step on a lot of open source projects to make it more clear what is actually being changed and if it actually is something we want to change. It might be a behavior that is actually ATOS working as intended. 

In the case of a bug the current behavior is the bug. I.e ATOS crashes when sending start to 5 objects etc.

Reviewer: Ok that makes sense!

* **What is the new behavior (if this is a feature change)?**

> * **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)

Reviewer: ```suggestion
* **Does the PR change the way the software in this project interfaces with internal/external components such as files, other software, etc?**
```

> * **Other information**:

Reviewer: Other information is "required information for a pull request"?

Author: Yeah that should have been as optional

Reviewer: Can you address this in the text?

Author: Yeah will add that and the other changes.
